back/app/models/permission.rb

```diff
@@ -50,6 +62,18 @@ def self.available_actions(permission_scope)
     ACTIONS[permission_scope&.participation_method]
   end
 
+  def self.enabled_actions(permission_scope)
+    # Remove any actions that are not enabled on the project
+    available_actions(permission_scope).filter_map do |action|
```

Comment:
I'm surprised you no longer have the pagination issue, because `filter_map` should also give you back an array. To keep using a scope, you could do something like `where.not(action: 'posting_idea') if !permission_scope&.posting_enabled?` etc.

Reply:
It's because this function gets called within the scope that generates the SQL instead of after the SQL is generated. So it will end up generating SQL in the scope something like `IN ('voting_idea', 'post_idea')`

Reply:
I'll add a couple of specs to cover as well

Reply:
Oh right, I didn't realize 😅 Yes, that definitely works!